### ISSUE TITLE
[DOC] Update pull request template to ask whether new tests have been added

### ISSUE THREAD

**Issue by achieveordie** [sktime/sktime#4066]
#### Describe the issue linked to the documentation

I suggest we update `sktime/PULL_REQUEST_TEMPLATE.md` and add another subsection asking the author if they have added any new tests. This may be more important for bug-fixing PRs but is also relevant to PRs related to code enhancements.

It is considered good practice to add tests with newly added code to enforce the fact that the code actually works. This will reduce the chance of introducing logical bugs which often escape tests that only check for spec compliance

This line will serve as a cue for the author to add tests to ensure that further changes to these files won't introduce the same kind of bug. 

#### Suggest a potential alternative/fix

``` md
#### Does your contribution introduce a new dependency? If yes, which one?

<!--
If your contribution does add a new hard dependency, we may suggest to initially develop your contribution in a separate companion package in https://github.com/sktime/ to keep external dependencies of the core sktime package to a minimum.
-->

#### Did you add any tests?

<! -- It is considered good practice to add tests whenever a new piece of code is added. While we don't mandate them, we encourage our contributors to add new tests to reduce the chance of introducing logical bugs. -->

#### What should a reviewer concentrate their feedback on?

<!-- This section is particularly useful if you have a pull request that is still in development. You can guide the reviews to focus on the parts that are ready for their comments. We suggest using bullets (indicated by * or -) and filled checkboxes [x] here -->
```

**Comment by achieveordie** [comment]
@miraep8 @fkiraly @GuzalBulatova

What do you think?

**Comment by fkiraly** [comment]
@achieveordie, good spot! We did not ask for this?

Initially, I thought maybe it is better to modify it a little, to sth similar to "is the change covered by tests?" or "is there a test certifying for the change" - but there may be a better formulation than that. Since automated tests may already cover the change.

But, on second thought, perhaps "did you add any tests" is clearest - since this is mostly the requirement for beginneres, and the advanced contributors will know what to do in the less frequent cases that are different.

**Comment by miraep8** [comment]
I also think it is a good idea!  And I like the phrasing you used along the lines "it is best practice".  I think the phrasing might be a bit wordy, but overall I like the change/think it would help new contributors! :)
